purchase_vendor_promotion/models/product_supplierinfo.py

```diff
@@ -22,3 +22,14 @@ def _check_promotion_dates(self):
                     raise ValidationError(
                         _("Promotion start date must be before end date.")
                     )
+
+    def _is_promotion_active_or_upcoming(self, date=None):
+        """
+        Consider Promotion is active if it's within start and end date or it's in the future
+        """
+        self.ensure_one()
+        if not self.is_promotion:
+            return False
+        if not date:
+            date = fields.Date.today()
+        return date <= self.date_start or (self.date_start <= date <= self.date_end)
```

Comment:
What if `date_start` and/or `date_end` are left empty?

Reply:
Hello @SilvioC2C, that is enforced by constraint `_check_promotion_dates`